Marlin/src/module/thermistor/thermistors.h

```diff
@@ -27,6 +27,8 @@
 #define THERMISTOR_TABLE_SCALE (HAL_ADC_RANGE / _BV(THERMISTOR_TABLE_ADC_RESOLUTION))
 #if ENABLED(HAL_ADC_FILTERED)
   #define OVERSAMPLENR 1
+#elif HAL_ADC_RESOLUTION > 10
+  #define OVERSAMPLENR (20 - HAL_ADC_RESOLUTION)
 #else
   #define OVERSAMPLENR 16
 #endif
```

Comment:
What does this change achieve. I'm not doubting it, Is this simply because fewer samples are needed to generate an averaged reading when using the higher resolution?

Is the number choice here arbitrary?

Reply:
In fact, this is a crutch so as not to change the existing oversampling mechanism. All measurements are summed in a 16-bit field, so a high ADC bit depth simultaneously with a large oversampling ratio will cause an overflow.
A few lines below there is a static validation for this (MAX_RAW_THERMISTOR_VALUE).

Since there is no easy way to transfer OVERSAMPLENR from hal (well, at least there are doubts that this needs to be introduced), I preferred to calculate it automatically for high-bit ADCs to avoid overflow. There is, of course, an opportunity to set HAL_ADC_FILTERED at the hal level, which will completely disable oversampling, but this is semantically incorrect, since no filtration is performed in this hal.

Reply:
Works for me. The main thing is to have at least some number of oversamples to deal with the natural variation in ADC readings. Guaranteed to be over 10 is reasonable.